pkg/fanal/secret/scanner.go

```diff
@@ -459,31 +459,33 @@ func findLocation(start, end int, content []byte) (int, int, types.Code, string)
 		lineEnd += start
 	}
 
-	match := string(content[start:end])
-	matchLine := string(content[lineStart:lineEnd])
-	if len(matchLine) > 100 {
+	var matchLine string
+	if lineEnd-lineStart > 100 {
 		truncatedLineStart := lo.Ternary(start-30 < 0, 0, start-30)
 		truncatedLineEnd := lo.Ternary(end+20 > len(content), len(content), end+20)
 		matchLine = string(content[truncatedLineStart:truncatedLineEnd])
+	} else {
+		matchLine = string(content[lineStart:lineEnd])
 	}
-	endLineNum := startLineNum + strings.Count(match, string(lineSep))
+	endLineNum := startLineNum + bytes.Count(content[start:end], lineSep)
```

Comment:
there is a moment is it correct to use `start:end` interval?

Reply:
could refer to this [reply](https://github.com/aquasecurity/trivy/pull/3453#discussion_r1085553095)